base/v0_6_exp/validate.go

```diff
@@ -41,7 +56,8 @@ func (rs Resource) Validate(c path.ContextPath) (r report.Report) {
 	if sources > 1 {
 		r.AddOnError(c.Append(field), common.ErrTooManyResourceSources)
 	}
-	return
+
+	return r
```

Comment:
I am pretty sure we dont need to return r explicitly here, we can keep it how it was.